master/buildbot/test/unit/test_changes_p4poller.py

```diff
@@ -344,6 +344,89 @@ def spawnProcess(pp, cmd, argv, env):
         self.assertEqual(
             self.changesource._ticket_passwd, 'TICKET_ID_GOES_HERE')
 
+    @defer.inlineCallbacks
+    def test_acquire_ticket_auth2(self):
+        self.attachChangeSource(
+            P4Source(p4port=None, p4user=None, p4passwd='pass',
+                     p4base='//depot/myproject/',
+                     split_file=lambda x: x.split('/', 1),
+                     use_tickets=True))
+        self.expectCommands(
+            gpo.Expect('p4', '-P', 'TICKET_ID_GOES_HERE',
+                       'changes', '-m', '1', '//depot/myproject/...').stdout(first_p4changes)
+        )
+
+        class FakeTransport:
```

Comment:
Could we extract this duplicated class to the top of the file? E.g. just above `class TestP4Poller`.

Reply:
Ah, yes. Just did so.